src/main/java/org/opensearch/neuralsearch/processor/CompoundTopDocs.java

```diff
@@ -88,17 +91,22 @@ public CompoundTopDocs(final TopDocs topDocs) {
             if (isHybridQueryDelimiterElement(scoreDoc) || isHybridQueryStartStopElement(scoreDoc)) {
                 ScoreDoc[] subQueryScores = scoreDocList.toArray(new ScoreDoc[0]);
                 TotalHits totalHits = new TotalHits(subQueryScores.length, TotalHits.Relation.EQUAL_TO);
-                TopDocs subQueryTopDocs = new TopDocs(totalHits, subQueryScores);
+                TopDocs subQueryTopDocs;
+                if (isSortEnabled && topDocs instanceof TopFieldDocs) {
```

Comment:
are we handling properly condition when `isSortEnabled == false` and  `topDocs instanceof TopFieldDocs`. Not sure if that's possible but if it is we probably should fail with runtime exception

Reply:
If sortEnabled = false then topdocs cannot be the instance of TopFieldDocs. Because creation of top field docs depends upon sortfields. No sort fields then it will be a sure shot top docs object and not an instance of top field docs. Creation of top docs is being handled by query phase.

Reply:
Then why are we doing "instance of"? It should be either just check for the `sortEnabled` or you need to handle both case when input object is and is not instance of. Currently in case it's not a TopFieldDocs the flow goes to TopDocs case, if that can handle such scenario (as it's a base class I think) I'm fine with it, otherwise we need to fail fast with Illegal Argument or something similar

Reply:
I have removed instance of check. I added after vijay's review. But it is not needed.